### ISSUE TITLE
ethereum::ChainId::default() bugs a bit

### ISSUE THREAD

**Issue by bonomat** [comit-network/comit-rs#2501]
# Problem

Right now, we support 3 different networks in `ethereum::ChainId`:
https://github.com/comit-network/comit-rs/blob/767ee5f0a94a7c61f6e617476866c2aa7483f2bc/cnd/src/swap_protocols/ledger/ethereum.rs#L32-L44
* mainnet -> 1
* ropsten -> 3
* regtest -> 17

This bit me now as we wanted to use Geth for our e2e tests. The default `chain_id` for Geth's dev chain is however `1337` - which I was using in the e2e tests. 
This would not be too bad as we also support arbitrary `chain_id`s in cnd. 
CI was constantly failing for some sanity tests which did not even need an ethereum node. 

The problem was that we do not specify a `chain_id` in the settings for these tests making cnd to default to `ethereum::ChainId::regtest()` ... which is `17`. 
Again, this is not a problem per se, however, in our CI, it may happen that we still have an ethereum node (Geth) running from other tests. This made cnd fail during start up because it tries to verify that the _configured_ `chain_id` (`regtest`) fits to the _actual_ `chain_id` (`1337` from Geth's dev net)

TL;DR;: 
Rethink `ethereum::ChainId::regtest()`

# Goal

Have _smart_ config defaults was our original plan. We should revisit this strategy and think about how we can support the special case Ethereum. 

# Note

We changed the default value of `ethereum::ChainId::regtest()` from `17` to `1337` in this PR: https://github.com/comit-network/comit-rs/pull/2499

# Personal opinion (not a recommendation)

Imho we should only default to a `chain_id` which is well-defined (i.e. one of the public networks), or move the `chain_id` into the specific chain setting (either Parity or Geth).

**Comment by bonomat** [comment]
It might be that this is will be obsolete with: https://github.com/comit-network/comit-rs/issues/1947
